osg-pilot-container/osgpilot_meter

```diff
@@ -0,0 +1,567 @@
+
+
+    """
+    create view pslot_totals as
+    select p.next_updated
+         , {pslot_cols}
+         , {dslot_cols}
+--       , count(*) as n_dslots
+      from inbox p
+      left join slots s
+        on p.name = s.pslot
+--     and p.SlotType = 'Partitionable'
+      left join inbox d
+        on d.name = s.dslot
+--     and d.SlotType = 'Dynamic'
+     group by p.name
+       ;
+    """.format(pslot_cols=fmt_coljoin("p.{}", want_pslot_attrs),
+               dslot_cols=fmt_coljoin("sum(d.{0}) as {0}", want_dslot_attrs)),
+
+
+    """
+    create view pslot_totals2 as
+    select p.next_updated
+         , {pslot_cols}
+         , {dslot_cols}
+--       , count(*) as n_dslots
+      from inbox p
+      left join inbox d
+        on p.MyAddress = d.MyAddress
+       and p.SlotId = d.SlotId
+       and p.SlotType = 'P'  -- 'Partitionable'
+       and d.SlotType = 'D'  -- 'Dynamic'
+     group by p.name
+       ;
+    """.format(pslot_cols=fmt_coljoin("p.{}", want_pslot_attrs),
+               dslot_cols=fmt_coljoin("sum(d.{0}) as {0}", want_dslot_attrs)),
+
+
+    """
+    create view updates as
+    select a.*
+         , b.name                      as b_name
+         , b.next_updated
+         , b.CPUsUsage                 as next_CPUsUsage
+         , b.MemoryUsage               as next_MemoryUsage
+         , next_updated - last_updated as elapsed_interval
+      from jobs a
+--    left join inbox b
+      left join pslot_totals b
+        on a.name = b.name;
+    """,
+
+
+    """
+    create view finished as
+    select *
+      from updates
+     where b_name is null;
+    """,
+
+
+    """
+    create view finished2 as
+    select DaemonStartTime                as StartTime
+         , last_updated                   as EndTime
+         , last_updated - DaemonStartTime as WallDuration
+         , total_cpu                      as CpuDuration
+         , Cpus                           as Processors
+         , MemoryUsage                    as Memory
+         , ifnull(GLIDEIN_Site,
+                  GLIDEIN_ResourceName)   as Site
+         , Machine                        as MachineName
+      from finished;
+    """,
+
+
+    """
+    create view updated as
+    select *
+         , next_CPUsUsage * elapsed_interval as next_cpu
+         , max(MemoryUsage,next_MemoryUsage) as next_mem
+      from updates
+     where b_name is not null;
+    """,
+
+
+    """
+    create view new_jobs as
+    select b.*
+--    from inbox b
+      from pslot_totals b
+      left join jobs a
+        on a.name = b.name
+     where a.name is null;
+    """,
+]
+
+
+# query collector for machine (startd) ads
+
+def query_current_attrs():
+    import htcondor
+    pool = 'flock.opensciencegrid.org' #:9618
+    coll = htcondor.Collector(pool)
+
+    filter_cond = 'SlotType != "Static"'
+    #filter_cond += ' && IsOsgVoContainer =?= True'
+
+    ads = coll.query(ad_type=htcondor.AdTypes.Startd, projection=want_attrs,
```

Comment:
Could you make `want_attrs` and your other constants cap-case? I was initially confused as to where it was defined.

```suggestion
    ads = coll.query(ad_type=htcondor.AdTypes.Startd, projection=WANT_ATTRS,
```